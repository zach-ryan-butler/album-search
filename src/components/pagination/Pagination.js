import React from 'react';
import PropTypes from 'prop-types';

function Pagination({ pageUp, pageDown, totalPages, page }) {
  const initialPage = totalPages === 0 ? page - 1 : page;

  return (
    <>
      <button disabled={page === 1} onClick={pageDown}>←</button>
      <span>page {initialPage} of {totalPages}</span>
      <button disabled={page === totalPages} onClick={pageUp}>→</button>
    </>
  );
}

Pagination.propTypes = {
  totalPages: PropTypes.number.isRequired,
  page: PropTypes.number.isRequired,
  pageUp: PropTypes.func.isRequired,
  pageDown: PropTypes.func.isRequired
};

export default Pagination;
