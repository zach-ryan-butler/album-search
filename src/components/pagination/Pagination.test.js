import React from 'react';
import { shallow } from 'enzyme';
import Pagination from './Pagination';

describe('Pagination component', () => {
  it('renders Pagination', () => {
    const wrapper = shallow(<Pagination 
      pageUp={() => {}}
      pageDown={() => {}}
      totalPages={10}
      page={5}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
