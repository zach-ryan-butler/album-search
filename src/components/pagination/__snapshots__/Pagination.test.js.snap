// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component renders Pagination 1`] = `
<Fragment>
  <button
    disabled={false}
    onClick={[Function]}
  >
    ←
  </button>
  <span>
    page 
    5
     of 
    10
  </span>
  <button
    disabled={false}
    onClick={[Function]}
  >
    →
  </button>
</Fragment>
`;
